client.go

```diff
@@ -1375,3 +1398,455 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+			EncryptedFileName: encryptedFileName,
+			Checksum:          checksum,
+			Size:              size,
+		}
+		uploadResp, err := file.UploadFile(uploadRequest)
+		if err != nil || uploadResp != 0 {
+			return createdSecret, "", err
+		}
+		// Register the file as being written
+		response, err := c.StorageClient.FileCommit(ctx, pendingFileURL.PendingFileID)
+		if err != nil {
+			return createdSecret, "", err
+		}
+		// get the file from the record
+		createdSecret = &pdsClient.Record{
+			Metadata: pdsClient.Meta{
+				RecordID:     response.Metadata.RecordID.String(),
+				WriterID:     response.Metadata.WriterID.String(),
+				UserID:       response.Metadata.UserID.String(),
+				Type:         response.Metadata.Type,
+				Plain:        response.Metadata.Plain,
+				Created:      response.Metadata.Created,
+				LastModified: response.Metadata.LastModified,
+				// storage client doesn't have a version
+				FileMeta: (*pdsClient.FileMeta)(response.Metadata.FileMeta),
+			},
+			Data:            response.Data,
+			RecordSignature: response.RecordSignature,
+		}
+	} else {
+		data := map[string]string{"secretValue": secret.SecretValue}
+		recordToWrite := pdsClient.WriteRecordRequest{
+			Data: data,
+			Metadata: pdsClient.Meta{
+				Type:     recordType,
+				WriterID: c.E3dbPDSClient.ClientID,
+				UserID:   c.E3dbPDSClient.ClientID,
+				Plain:    plain,
+			},
+		}
+		encryptedRecord, err := c.E3dbPDSClient.EncryptRecord(ctx, recordToWrite)
+		if err != nil {
+			return createdSecret, "", err
+		}
+		createdSecret, err = c.E3dbPDSClient.WriteRecord(ctx, encryptedRecord)
+		if err != nil {
+			return createdSecret, "", err
+		}
+	}
+	keyReq := pdsClient.GetOrCreateAccessKeyRequest{
+		WriterID:   c.E3dbPDSClient.ClientID,
+		UserID:     c.E3dbPDSClient.ClientID,
+		ReaderID:   c.E3dbPDSClient.ClientID,
+		RecordType: recordType,
+	}
+	keyResp, err := c.E3dbPDSClient.GetOrCreateAccessKey(ctx, keyReq)
+	if err != nil {
+		return createdSecret, "", err
+	}
+	wrappedAK, err := EncryptAccessKeyForGroup(*c.StorageClient, keyResp)
+	if err != nil {
+		return createdSecret, "", err
+	}
+	accessKeyReq := storageClient.GroupAccessKeyRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: wrappedAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, encryptedGroupAK, err := c.StorageClient.CreateGroupAccessKey(ctx, accessKeyReq)
+	if err != nil {
+		return createdSecret, "", err
+	}
+	// share secret with group
+	secretShareReq := storageClient.ShareGroupRecordRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: encryptedGroupAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, err = c.StorageClient.ShareRecordWithGroup(ctx, secretShareReq)
+	if err != nil {
+		return createdSecret, "", err
+	}
+	return createdSecret, "", nil
+}
+
+func EncryptAccessKeyForGroup(member storageClient.StorageClient, ak e3dbClients.SymmetricKey) (string, error) {
```

Comment:
Can you double check to see if there is a way to use any of the existing methods for managing group membership keys for this please

https://github.com/tozny/e3db-clients-go/blob/master/storageClient/storageClient.go#L216
https://github.com/tozny/e3db-clients-go/blob/master/storageClient/storageClient.go#L237
https://github.com/tozny/e3db-clients-go/blob/master/storageClient/storageClient.go#L282